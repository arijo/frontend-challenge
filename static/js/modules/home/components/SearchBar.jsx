import lo from 'lodash';
import React from 'react';
import {Col, Row} from 'react-grid-system';
import style from '../../../../css/modules/home/Home.scss';
import store from '../../../store';
import {fetchMovies} from '../actions';

class SearchBar extends React.Component {
    constructor(props) {
        super(props);
        this.doSearch = lo.debounce(this.doSearch.bind(this), 1000, {});
    }
    doSearch(value) {
        store.dispatch(fetchMovies(value))
    }
    render() {
        return (
            <div>
                <Row>
                    <Col sm={12}>
                        <div className={style.searchBar}>
                            <span className={style.magnifier}></span>
                            <input onKeyUp={(e) => this.doSearch(e.target.value)} placeholder={'Search movies ...'}/>
                        </div>
                    </Col>
                </Row>
            </div>
        )
    }
}

export default SearchBar;
