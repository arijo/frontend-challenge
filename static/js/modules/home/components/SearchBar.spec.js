import SearchBar from './SearchBar.jsx';

it('renders correctly', () => {
    const wrapper = shallow(
        <SearchBar/>
    );
    expect(wrapper).toMatchSnapshot();
})