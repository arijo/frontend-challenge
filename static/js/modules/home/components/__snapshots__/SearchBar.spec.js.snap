// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <Row
    align="normal"
    component="div"
    debug={false}
    justify="start"
    nogutter={false}
    style={Object {}}
  >
    <Col
      component="div"
      debug={false}
      lg={null}
      md={null}
      offset={Object {}}
      pull={Object {}}
      push={Object {}}
      sm={12}
      style={Object {}}
      xl={null}
      xs={null}
    >
      <div
        className="searchBar"
      >
        <span
          className="magnifier"
        />
        <input
          onKeyUp={[Function]}
          placeholder="Search movies ..."
        />
      </div>
    </Col>
  </Row>
</div>
`;
